src/Parser/DeathKnight/Unholy/Modules/Features/VirulentPlagueEfficiency.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import Enemies from 'Parser/Core/Modules/Enemies';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import { encodeTargetString } from 'Parser/Core/Modules/EnemyInstances';
+import Wrapper from 'common/Wrapper';
+
+class VirulentPlagueEfficiency extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+	enemies: Enemies,
+  };
+  
+  targets = {};
+  
+  totalOutBreakCasts = 0;
+  totalVirulentDuration = 0;
+  totalTimeWasted = 0;
+  
+  get VirulentDuration(){
+	if (this.combatants.selected.hasTalent(SPELLS.EBON_FEVER_TALENT.id)) {
+		return(19.65);
+	} else{
+		return(33.3);
+	}
+  }
+  
+  on_byPlayer_refreshdebuff(event){
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.VIRULENT_PLAGUE.id){
+		this.targets[encodeTargetString(event.targetID, event.targetInstance)] = event.timestamp;
```

Comment:
I think you might have mixed up `refreshdebuff` and `applydebuff`, the pandemic duration should be added here in `refreshdebuff`. I would suggest that you save a timestamp for when you expect the dot to expire instead of when it was last cast, that way you can also account for pandemic effects properly: 
```
const remainingDuration = this.targets[encodeTargetString(event.targetID, event.targetInstance)] - event.timestamp;
if (remainingDuration < 1000*0.3*this.VirulentDuration) { //less than full pandemic
  this.targets[encodeTargetString(event.targetID, event.targetInstance)] = event.timestamp + this.VirulentDuration + remainingDuration;
} else { //full pandemic
  this.targets[encodeTargetString(event.targetID, event.targetInstance)] = event.timestamp + this.VirulentDuration * 1.3;
}
```

and then in `applydebuff` you just do `this.targets[encodeTargetString(event.targetID, event.targetInstance)] = event.timestamp + this.VirulentDuration;`